caffe2/CMakeLists.txt

```diff
@@ -1162,6 +1162,18 @@ if(USE_DISTRIBUTED)
     target_compile_definitions(torch_cpu PRIVATE
       USE_TENSORPIPE
     )
+    # This is needed for torch/csrc/distributed/rpc/macros.h to work
+    # in libtorch_distributed_sources source files, e.g., rref_impl.cpp.
+    # See tools/build_variables.bzl
+    if(USE_CUDA)
+      target_compile_definitions(torch_cpu PUBLIC USE_CUDA)
+    endif()
```

Comment:
This is wrong. torch_cpu library should be CUDA indepentent (i.e. torch_cpu from pure CPU build and torch_cpu from CUDA build could be used as a drop replacement)